atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/AtlasRestoreService.java

```diff
@@ -173,79 +182,89 @@ public Set<Namespace> visitUnknown(String unknownType) {
             log.warn(
-                    "Completed backups were not found for some namespaces",
-                    SafeArg.of("namespacesWithBackup", namespacesWithBackup),
-                    SafeArg.of("namespacesWithoutBackup", namespacesWithoutBackup));
+                    "Completed backups were not found for some atlasServices",
+                    SafeArg.of("atlasServicesWithBackup", atlasServicesWithBackup),
+                    SafeArg.of("atlasServicesWithoutBackup", atlasServicesWithoutBackup));
         }
 
         // Fast forward timestamps
         Map<Namespace, CompletedBackup> completeRequest = KeyedStream.stream(completedBackups)
-                .mapKeys(RestoreRequest::newNamespace)
+                .mapKeys(RestoreRequest::newAtlasService)
+                .mapKeys(AtlasService::getNamespace)
+                .collectToMap();
+        Map<Namespace, AtlasService> namespaceToServices = KeyedStream.of(atlasServicesToRestore)
+                .mapKeys(AtlasService::getNamespace)
                 .collectToMap();
+
         CompleteRestoreResponse response =
                 atlasRestoreClient.completeRestore(authHeader, CompleteRestoreRequest.of(completeRequest));
-        Set<Namespace> successfulNamespaces = response.getSuccessfulNamespaces();
-        Set<Namespace> failedNamespaces = Sets.difference(namespacesToRestore, successfulNamespaces);
-        if (!failedNamespaces.isEmpty()) {
+        Set<AtlasService> successfulAtlasServices = response.getSuccessfulNamespaces().stream()
+                .map(namespaceToServices::get)
+                .collect(Collectors.toSet());
+        Set<AtlasService> failedAtlasServices = Sets.difference(atlasServicesToRestore, successfulAtlasServices);
+        if (!failedAtlasServices.isEmpty()) {
             log.error(
-                    "Failed to fast-forward timestamp for some namespaces. These will not be re-enabled.",
-                    SafeArg.of("failedNamespaces", failedNamespaces),
-                    SafeArg.of("fastForwardedNamespaces", successfulNamespaces));
+                    "Failed to fast-forward timestamp for some atlasServices. These will not be re-enabled.",
+                    SafeArg.of("failedAtlasServices", failedAtlasServices),
+                    SafeArg.of("fastForwardedAtlasServices", successfulAtlasServices));
         }
 
         // Re-enable timelock
-        Set<Namespace> newNamespaces =
-                restoreRequests.stream().map(RestoreRequest::newNamespace).collect(Collectors.toSet());
         timeLockManagementService.reenableTimelock(
-                authHeader, ReenableNamespacesRequest.of(successfulNamespaces, backupId));
-        if (successfulNamespaces.containsAll(newNamespaces)) {
+                authHeader, ReenableNamespacesRequest.of(response.getSuccessfulNamespaces(), backupId));
+        if (successfulAtlasServices.containsAll(atlasServicesToRestore)) {
```

Comment:
if this is not the case, do we want to log something else, or do we have sufficient logging for the possible failure cases?

Reply:
We already log above, but I can combine the two into an if/else for clarity